pkg/types/intoto/v0.0.1/entry.go

```diff
@@ -150,6 +151,18 @@ func (v *V001Entry) Canonicalize(ctx context.Context) ([]byte, error) {
 			},
 		},
 	}
+	attestation := v.Attestation()
+	if attestation != nil {
+		decodedAttestation, err := base64.StdEncoding.DecodeString(string(attestation))
+		if err != nil {
+			return nil, errors.Wrap(err, "decoding attestation")
+		}
+		attH := sha256.Sum256(decodedAttestation)
```

Comment:
Should you either return an error if a non sha256 algorithm enum is specified?

Reply:
we're setting the value, its not user selectable

Reply:
Ah, looks good then!